pkg/cloud/services/compute/instance.go

```diff
@@ -194,54 +175,31 @@ func createInstance(is *Service, clusterName string, i *infrav1.Instance) (*infr
 			}
 		}
 
-		portsList = append(portsList, servers.Network{
+		portList = append(portList, servers.Network{
 			Port: port.ID,
 		})
-
-		if i.Trunk {
-			allPages, err := trunks.List(is.networkClient, trunks.ListOpts{
-				Name:   i.Name,
-				PortID: port.ID,
-			}).AllPages()
-			if err != nil {
-				return nil, fmt.Errorf("searching for existing trunk for server err: %v", err)
-			}
-			trunkList, err := trunks.ExtractTrunks(allPages)
-			if err != nil {
-				return nil, fmt.Errorf("searching for existing trunk for server err: %v", err)
-			}
-			var trunk trunks.Trunk
-			if len(trunkList) == 0 {
-				// create trunk with the previous port as parent
-				trunkCreateOpts := trunks.CreateOpts{
-					Name:   i.Name,
-					PortID: port.ID,
-				}
-				newTrunk, err := trunks.Create(is.networkClient, trunkCreateOpts).Extract()
-				if err != nil {
-					return nil, fmt.Errorf("create trunk for server err: %v", err)
-				}
-				trunk = *newTrunk
-			} else {
-				trunk = trunkList[0]
-			}
-
-			_, err = attributestags.ReplaceAll(is.networkClient, "trunks", trunk.ID, attributestags.ReplaceAllOpts{
-				Tags: i.Tags,
-			}).Extract()
-			if err != nil {
-				return nil, fmt.Errorf("tagging trunk for server err: %v", err)
-			}
-		}
 	}
 
 	if i.Subnet != "" && accessIPv4 == "" {
-		if errd := deletePorts(is, portsList); errd != nil {
-			return nil, fmt.Errorf("no ports with fixed IPs found on Subnet %q: error cleaning up ports: %v", i.Subnet, errd)
+		if errd := deletePorts(is, object, portList); errd != nil {
+			return nil, errd
 		}
 		return nil, fmt.Errorf("no ports with fixed IPs found on Subnet %q", i.Subnet)
 	}
 
+	instance, err := createServer(is, object, portList, accessIPv4, i)
+	if err != nil {
+		return nil, err
+	}
+	return instance, nil
+}
+
+func createServer(is *Service, object runtime.Object, portList []servers.Network, accessIPv4 string, i *infrav1.Instance) (*infrav1.Instance, error) {
+	imageID, err := getImageID(is, i.Image)
+	if err != nil {
+		return nil, fmt.Errorf("create new server err: %v", err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("create new server: %v", err)
```
Can you drop those `err` strings everywhere, please?